sql/resource_alert.go

```diff
@@ -0,0 +1,125 @@
+		common.CustomizeSchemaPath(m, "condition", "operand", "column", "name").SetRequired()
+		common.CustomizeSchemaPath(m, "condition", "empty_result_state").SetValidateFunc(
+			validation.StringInSlice([]string{"UNKNOWN", "OK", "TRIGGERED"}, true))
+		// We may not need it for some conditions
+		// common.CustomizeSchemaPath(m, "condition", "threshold").SetRequired()
+		common.CustomizeSchemaPath(m, "condition", "threshold", "value").SetRequired()
+		alof := []string{
+			"condition.0.threshold.0.value.0.string_value",
+			"condition.0.threshold.0.value.0.double_value",
+			"condition.0.threshold.0.value.0.bool_value",
+		}
+		for _, f := range alof {
+			common.CustomizeSchemaPath(m, "condition", "threshold", "value",
+				strings.TrimPrefix(f, "condition.0.threshold.0.value.0.")).SetExactlyOneOf(alof)
+		}
+		common.CustomizeSchemaPath(m, "owner_user_name").SetSuppressDiff()
+		common.CustomizeSchemaPath(m, "id").SetReadOnly()
+		common.CustomizeSchemaPath(m, "create_time").SetReadOnly()
+		common.CustomizeSchemaPath(m, "lifecycle_state").SetReadOnly()
+		common.CustomizeSchemaPath(m, "state").SetReadOnly()
+		common.CustomizeSchemaPath(m, "trigger_time").SetReadOnly()
+		common.CustomizeSchemaPath(m, "update_time").SetReadOnly()
+		return m
+	})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var a sql.CreateAlertRequestAlert
+			common.DataToStructPointer(d, s, &a)
+			apiAlert, err := w.Alerts.Create(ctx, sql.CreateAlertRequest{
+				Alert: &a,
+			})
+			if err != nil {
+				return err
+			}
+			d.SetId(apiAlert.Id)
+			owner := d.Get("owner_user_name").(string)
+			if owner != "" {
+				_, err = w.Alerts.Update(ctx, sql.UpdateAlertRequest{
+					Alert: &sql.UpdateAlertRequestAlert{
+						OwnerUserName: owner,
+					},
+					Id:         apiAlert.Id,
+					UpdateMask: "owner_user_name",
+				})
+			}
+			return err
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			apiAlert, err := w.Alerts.GetById(ctx, d.Id())
+			if err != nil {
+				log.Printf("[WARN] error getting alert by ID: %v", err)
+				return err
+			}
+			parentPath := d.Get("parent_path").(string)
+			if parentPath != "" && strings.HasPrefix(apiAlert.ParentPath, "/Workspace") && !strings.HasPrefix(parentPath, "/Workspace") {
+				apiAlert.ParentPath = strings.TrimPrefix(parentPath, "/Workspace")
+			}
+			return common.StructToData(apiAlert, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var a sql.UpdateAlertRequestAlert
+			updateMask := "display_name,query_id,seconds_to_retrigger,condition,custom_body,custom_subject"
```

Comment:
Do we use updateMask anywhere else in the TF provider? I wonder if we should always set it for all fields, even if fields don't have any change, since the `*schema.ResourceData` has the desired values for all of those fields. Mainly asking from the perspective of if we need logic around setting update masks or if we can avoid it to make things simpler.